Manual/NotationsMacros.lean

```diff
@@ -504,32 +504,428 @@ Left-associative operators use the higher precedence for their right argument, w
+{TODO}[Default precedence]
+Just as with operators and notations, syntax rules may be manually provided with a name; if they are not, an otherwise-unused name is generated.
+Whether provided or generated, this name is used as the syntax kind in the resulting {name Lean.Syntax.node}`node`.
+
+The body of a syntax declaration is even more flexible than that of a notation.
+String literals specify atoms to match.
+Subterms may be drawn from any syntax category, rather than just terms, and they may be optional or repeated, with or without interleaved comma separators.
+Identifiers in syntax rules indicate syntax categories, rather than naming subterms as they do in notations.
+
+
+Finally, the syntax rule specifies which syntax category it extends.
+It is an error to declare a syntax rule in a nonexistent category.
+
+```lean (show := false)
+-- verify preceding para
+/-- error: unknown category 'nuhUh' -/
+#guard_msgs in
+syntax "blah" : nuhUh
+```
+
+
+:::syntax stx (open := false)
+The syntactic category `stx` is the grammar of specifiers that may occur in the body of a {keywordOf Lean.Parser.Command.syntax}`syntax` command.
+
+String literals are parsed as {tech}[atoms] (including both keywords such as `if`, `#eval`, or `where`):
+```grammar
+$s:str
+```
+Leading and trailing spaces in the strings do not affect parsing, but they cause Lean to insert spaces in the corresponding position when displaying the syntax in {tech}[proof states] and error messages.
+Ordinarily, valid identifiers occurring as atoms in syntax rules become reserved keywords.
+Preceding a string literal with an ampersand (`&`) suppresses this behavior:
+```grammar
+&$s:str
+```
+
+Identifiers specify the syntactic category expected in a given position, and may optionally provide a precedence:{TODO}[Default prec here?]
+```grammar
+$x:ident$[:$p]?
+```
+
+The `*` modifier is the Kleene star, matching zero or more repetitions of the preceding syntax:
+```grammar
+$s:stx *
+```
+The `+` modifier matches one or more repetitions of the preceding syntax:
+```grammar
+$s:stx +
+```
+The `?` modifier makes a subterm optional, and matches zero or one, but not more, repetitions of the preceding syntax:
+```grammar
+$s:stx ?
+```
+
+The `,*` modifier matches zero or more repetitions of the preceding syntax with interleaved commas:
+```grammar
+$_:stx ,*
+```
+
+The `,+` modifier matches one or more repetitions of the preceding syntax with interleaved commas:
+```grammar
+$_:stx ,+
+```
+
+The `,*,?` modifier matches zero or more repetitions of the preceding syntax with interleaved commas, allowing an optional trailing comma after the final repetition:
+```grammar
+$_:stx ,*,?
+```
+
+The `,*,?` modifier matches one or more repetitions of the preceding syntax with interleaved commas, allowing an optional trailing comma after the final repetition:
```

Comment:
this should be `,+,?`. you repeated the `*`

Reply:
Thanks!